galasa-ui/src/components/test-runs/TestRunsTabs.tsx

```diff
@@ -27,106 +27,123 @@ interface TestRunsTabProps {
   resultsNamesPromise: Promise<string[]>;
 }
 
-
 export default function TestRunsTabs({runsListPromise, requestorNamesPromise, resultsNamesPromise}: TestRunsTabProps) {
   const translations = useTranslations("TestRunsTabs");
   const router = useRouter();
   const pathname = usePathname();
   const searchParams = useSearchParams();
-  const [selectedVisibleColumns, setSelectedVisibleColumns] = useState<string[]>([
-    COLUMNS_IDS.SUBMITTED_AT, 
-    COLUMNS_IDS.TEST_RUN_NAME,
-    COLUMNS_IDS.REQUESTOR, 
-    COLUMNS_IDS.TEST_NAME,
-    COLUMNS_IDS.STATUS, 
-    COLUMNS_IDS.RESULT
-  ]);
 
-  const [columnsOrder, setColumnsOrder] = useState<{ id: string; columnName: string }[]>(RESULTS_TABLE_COLUMNS);
-  const [isInitialized, setIsInitialized] = useState(false);
+  const TABS_IDS = ['timeframe', 'table-design', 'search-criteria', 'results'];
 
-  // Load from URL parameters first (only on mount)
-  useEffect(() => {
-    const visibleColumnsParam = searchParams.get("visibleColumns");
-    const columnsOrderParam = searchParams.get("columnsOrder");
+  // Initialize selectedIndex based on URL parameters or default to first tab
+  const [selectedIndex, setSelectedIndex] = useState(() => {
+    const tabParam = searchParams.get("tab");
+    const initialIndex = tabParam ? TABS_IDS.indexOf(tabParam) : -1;
+    return initialIndex !== -1 ? initialIndex : 0;
+  });
```

Comment:
Instead of indexes, would it be easier to track the selected tab by its ID?

So instead of a list of tabs, we could use a map of key-value pairs where the keys are the IDs (e.g. 'timeframe', 'table-design' etc, and the values are the tab labels. Just a thought!

Reply:
No problem, happy to keep it as-is in that case!